binary_search.py

```diff
@@ -0,0 +1,61 @@
+class BinarySearch():
+  """
+  Hashing and binary search both address the same problem: finding an item in a collection. What are some trade-offs between the two strategies?
+  * You can only use binary search on something you're able to sort by some sort of metric. You can use hashing on almost anything, possibly? since theoretically you could use pointers to point to whatever it is. and if it's a whole object perhaps you hash the memory location? I'm not sure, actually, and now that I mention it I wonder how that is done for languages that allow say, values to be entire objects.
```

Comment:
This is correct.

In Java for example (where the key can be an entire object), you're expected to define a custom `hashCode` function, which typically looks at the different fields of the object in order to generate the hash value.